internal/controller/controllers/agent_reclaimer.go

```diff
@@ -74,68 +134,45 @@ func (r *agentReclaimer) ensureSpokeAgentSecret(ctx context.Context, c client.Cl
 		}
 	}
 
-	key := types.NamespacedName{
-		Name:      spokeReclaimSecretName(infraEnvID),
-		Namespace: spokeReclaimNamespaceName,
-	}
 	secret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      key.Name,
-			Namespace: key.Namespace,
+			Name:      spokeReclaimSecretName(infraEnvID),
+			Namespace: spokeReclaimNamespaceName,
 		},
-		Type: corev1.SecretTypeOpaque,
 	}
-	err := c.Get(ctx, key, secret)
-	if err != nil && !k8serrors.IsNotFound(err) {
-		return errors.Wrapf(err, "failed to get secret %s/%s", key.Namespace, key.Name)
-	}
-
-	if k8serrors.IsNotFound(err) {
+	mutate := func() error {
+		secret.Type = corev1.SecretTypeOpaque
 		secret.Data = map[string][]byte{"auth-token": []byte(authToken)}
-		if err := c.Create(ctx, secret); err != nil {
-			return errors.Wrapf(err, "failed to create secret %s", spokeReclaimNamespaceName)
-		}
+		return nil
 	}
-
-	return nil
+	_, err := controllerutil.CreateOrUpdate(ctx, c, secret, mutate)
```

Comment:
We don't want to have a log if the result is not none (unchanged)?

Reply:
:shrug: We could but I'm not sure it matters.
In that case we're doing what we're supposed to do.

But if you would prefer a log I can add one.

Reply:
Not very crucial, but this is the behavior on other controllers, so thought it worth adding it for consistency.